update search/heuristic (#91)
OI-wiki/i18n: docs/search/heuristic.md

## Introduction

> In simple terms, heuristic search is a technique that analyzes both taking and non-taking operations, and select a better solution (or delete the invalid solution).

Reviewer: ```suggestion
In simple terms, heuristic search is a technique that analyzes the results of both performing and not performing operations, and select a better solution or remove the invalid solutions.
```

Since the concept is too abstract, we will take a few examples to explain.

??? note " Sample problem [「NOIP2005 general level」Gather herbs](https://www.luogu.com.cn/problem/P1048) (original link in Chinese)"

>     Problem: There are $N$ items and a backpack with a capacity of $W$ . Each item has two attributes: weight $wi$ and value $vi$ . It is required to select several items (each item can only be selected once) to put in the backpack to maximize the total value and the total weight of the items while not exceeding the capacity of the backpack.

Reviewer: 没说要 (maximize) the total weight of the items 啊 🌚

## Code

> We write an evaluation function $f$ that can cut all invalid 0 branches (that is, cut a large number of useless branches without selecting).

Reviewer: ```suggestion
We write an evaluation function $f$ that can cut off all invalid 0 branches (that is, cut off myriad of useless branches without selecting).
```
剪枝和下面统一一下表述 
另外优化格式好评 :+1:

> The running process of the evaluation function $f$ is as follows:

Reviewer: ```suggestion
The process of the evaluation function $f$ is as follows:
```

> When taking the items, we need to check whether it exceeds the specified volume (feasible pruning).

Reviewer: ```suggestion
When taking an item, check whether it exceeds the capacity (feasible pruning).
```

> When not taking it, check whether the value of the remaining medicine + the existing value is greater than the optimal solution found so far (optimal pruning).

Reviewer: 两行物品的单复数统一一下。

> When not taking it, check whether the value of the sum of remaining medicine and existing value is greater than the optimal solution found so far (optimal pruning).

Reviewer: ```suggestion
When not taking it, check whether the sum of the value of remaining medicine plus existing value is greater than the optimal solution found so far (optimal pruning).
```
